📖 Update Try-it page based on the v1beta1 API (#224)
metal3-io/metal3-io.github.io: try-it.md

### 1.1. Prerequisites

> - System with CentOS 8 or Ubuntu 20.04

Reviewer: ```suggestion
- System with Centos 8 Stream or Ubuntu 20.04
```

- Bare metal preferred, as we will be creating VMs to emulate bare metal hosts
- Run as a user with passwordless sudo access
- Minimum resource requirements for the host machine: 4C CPUs, 16 GB RAM memory

> - Required dependencies: make, kubelet, kubeadm, kubectl, go (version 1.17.5)

Reviewer: I would suggest not to include these as dependencies since most of them will be installed by m3-dev-env scripts

### 1.1. Prerequisites

- System with CentOS 8 or Ubuntu 20.04
- Bare metal preferred, as we will be creating VMs to emulate bare metal hosts
- Run as a user with passwordless sudo access
- Minimum resource requirements for the host machine: 4C CPUs, 16 GB RAM memory

> - The host machine should have a floating IP

Reviewer 2: I think this is not necessary but please correct me if I'm wrong. It should be possible to run all of this locally without any floating IP.

This section describes how to trigger provisioning of a cluster and hosts via
`Machine` objects as part of the Cluster API integration. This uses Cluster API
[v1beta1](https://github.com/kubernetes-sigs/cluster-api/tree/v1.0.2) and
assumes that metal3-dev-env is deployed with the environment variable

> **CAPM3_VERSION** set to **v1beta1**. This is the default behavior. The v1beta1 deployment can be done with
> Ubuntu 18.04 or Centos 8 target host images. Please make sure to meet [resource requirements](#11-prerequisites) for successful deployment:

Reviewer 3: ```suggestion
**CAPM3_VERSION** set to **v1beta1**. This is the default behavior. The v1beta1 deployment can be done with
Ubuntu 20.04 or Centos 8 target host images. Please make sure to meet [resource requirements](#11-prerequisites) for successful deployment:
```

> Ubuntu 20.04 or Centos 8 target host images. Please make sure to meet [resource requirements](#11-prerequisites) for successful deployment:

Reviewer: ```suggestion
Ubuntu 20.04 or Centos 8 Stream target host images. Please make sure to meet [resource requirements](#11-prerequisites) for successful deployment:
```

This section describes how to trigger provisioning of a cluster and hosts via
`Machine` objects as part of the Cluster API integration. This uses Cluster API

> <<<<<<< HEAD

Reviewer 4: should be fixed

First, you must scale down the deployment of the `baremetal-operator` running
in the cluster.

```console
$ kubectl scale deployment baremetal-operator-controller-manager -n baremetal-operator-system --replicas=0
```

To be able to run `baremetal-operator` locally, you need to install
[operator-sdk](https://github.com/operator-framework). After that, you can run
the `baremetal-operator` including any custom changes.

```console
$ cd ~/go/src/github.com/metal3-io/baremetal-operator
$ make run
```

### 2.5. Running Custom Cluster API Provider Metal3

There are two Cluster API related managers running in the cluster. One
includes set of generic controllers, and the other includes a custom Machine
controller for Metal3. You can try changes to

> `cluster-api-provider-metal3` and the changes will be automaticly applied when using Tilt.

Reviewer 3: we haven't talked about Tilt until this point and not sure if the mention of it will not confuse people.
I think either we need to explain how to work with Tilt/give a reference to a document where we have explained or
keep this part as it is

First, you must scale down the deployment of the `baremetal-operator` running
in the cluster.

```console
$ kubectl scale deployment baremetal-operator-controller-manager -n baremetal-operator-system --replicas=0
```

To be able to run `baremetal-operator` locally, you need to install
[operator-sdk](https://github.com/operator-framework). After that, you can run
the `baremetal-operator` including any custom changes.

```console
$ cd ~/go/src/github.com/metal3-io/baremetal-operator
$ make run
```

### 2.5. Running Custom Cluster API Provider Metal3

There are two Cluster API related managers running in the cluster. One
includes set of generic controllers, and the other includes a custom Machine
controller for Metal3.

#### Tilt development environment

[Tilt](https://tilt.dev) setup can deploy CAPM3 in a local kind cluster. Since
Tilt is applied in the metal3-dev-env deployment, you can make changes inside
the `cluster-api-provider-metal3` folder and Tilt will deploy the changes
automatically.

> If you deployed CAMP3 separately and want to make changes in CAPM3, then

Reviewer: ```suggestion
If you deployed CAPM3 separately and want to make changes to it, then
```

First, you must scale down the deployment of the `baremetal-operator` running
in the cluster.

```console
$ kubectl scale deployment baremetal-operator-controller-manager -n baremetal-operator-system --replicas=0
```

To be able to run `baremetal-operator` locally, you need to install
[operator-sdk](https://github.com/operator-framework). After that, you can run
the `baremetal-operator` including any custom changes.

```console
$ cd ~/go/src/github.com/metal3-io/baremetal-operator
$ make run
```

### 2.5. Running Custom Cluster API Provider Metal3

There are two Cluster API related managers running in the cluster. One
includes set of generic controllers, and the other includes a custom Machine
controller for Metal3.

#### Tilt development environment

[Tilt](https://tilt.dev) setup can deploy CAPM3 in a local kind cluster. Since

> Tilt is applied in the metal3-dev-env deployment, you can try changes to
> `cluster-api-provider-metal3` and Tilt will deploy the changes automaticly.

Reviewer: ```suggestion
Tilt is applied in the metal3-dev-env deployment, you can make changes inside the
`cluster-api-provider-metal3` folder and Tilt will deploy the changes automatically.
```

in the cluster.

```sh
$ kubectl scale deployment baremetal-operator-controller-manager -n baremetal-operator-system --replicas=0
```

To be able to run `baremetal-operator` locally, you need to install
[operator-sdk](https://github.com/operator-framework). After that, you can run
the `baremetal-operator` including any custom changes.

```sh
$ cd ~/go/src/github.com/metal3-io/baremetal-operator

> $ make run

Reviewer 2: No big deal, but when including the `$` prompt, it would be more correct to use `console` or `shell` instead of `sh`. I noticed that this guide is using `sh` all the time and I think it would be nice to change it to `console`. It makes the syntax highlighting correct.

See https://stackoverflow.com/a/49004070 .

```

### 2.5. Running Custom Cluster API Provider Metal3

There are two Cluster API related managers running in the cluster. One
includes set of generic controllers, and the other includes a custom Machine
controller for Metal3. You can try changes to

> `cluster-api-provider-metal3` and the changes will be automaticly applied.

Reviewer 2: This sounds strange. Is it when using Tilt? In that case I think it should be mentioned. E.g. "... changes will be automatically applied when using Tilt."

```sh
[notstack@metal3 metal3-dev-env]$ export CONTAINER_RUNTIME=docker
[notstack@metal3 metal3-dev-env]$ baremetal node list
+--------------------------------------+---------------+--------------------------------------+-------------+--------------------+-------------+
| UUID                                 | Name          | Instance UUID                        | Power State | Provisioning State | Maintenance |
+--------------------------------------+---------------+--------------------------------------+-------------+--------------------+-------------+

> | b423ee9c-66d8-48dd-bd6f-656b93140504 | metal3~node-1 | 7f51f14b-7701-436a-85ba-7dbc7315b3cb | power off   | available          | False       |
> | 882533c5-2f14-49f6-aa44-517e1e404fd8 | metal3~node-0 | 14fbcd25-4d09-4aca-9628-a789ba3e175c | power off   | available          | False       |

Reviewer 5: ```suggestion
| b423ee9c-66d8-48dd-bd6f-656b93140504 | node-1 | 7f51f14b-7701-436a-85ba-7dbc7315b3cb | power off   | available          | False       |
| 882533c5-2f14-49f6-aa44-517e1e404fd8 | node-0 | 14fbcd25-4d09-4aca-9628-a789ba3e175c | power off   | available          | False       |
```

Reviewer 5: not sure if there is a change to the naming of bmhs
